### PR TITLE
FOGL-1433 Revamp configuration page [Tree structure] 

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<div id="add-category-child" class="modal">
+  <div class="modal-background"></div>
+  <div class="modal-card">
+    <header class="modal-card-head">
+      <p class="modal-card-title is-size-5">Add Category Child</p>
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.html]: Add Child Category

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  constructor(
+    private configService: ConfigurationService,
+    private alertService: AlertService
+  ) {}
+
+  ngOnInit() {
+    this.categoryData = {
+      parentCategory: '',
+      child: ''
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts]: childCategory ? to keep it consistent with `parentCategory` or just parent/ child

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    };
+  }
+
+  public setCategoryData(key) {
+    this.categoryData = {
+      parentCategory: key
+    };
+  }
+
+  public toggleModal(isOpen: Boolean, form: NgForm = null) {
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts]: `toggle<WHICH>Modal`

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const modal = <HTMLDivElement>document.getElementById('add-category-child');
+    if (isOpen) {
+      modal.classList.add('is-active');
+      return;
+    }
+    modal.classList.remove('is-active');
+  }
+
+  public resetAddChildForm(form: NgForm) {
+    form.resetForm();
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts]: do we really need this as a function?

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  public addChild(form: NgForm) {
+    const parent = form.controls['parentCategory'].value;
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts]: no progress bar?

File: src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    const child = [];
+    child.push(form.controls['child'].value) ;
+    this.configService
+      .addChild(parent, child)
+      .subscribe(() => {
+          this.notify.emit(this.categoryData);
+          this.toggleModal(false, null);
+          this.alertService.success('Children has been added successfully');
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/add-category-child/add-category-child.component.ts]: Child has been / Children have been

File: src/app/components/core/configuration-manager/configuration-manager.component.css [hunk 7]
```diff
@@ -0,0 +1,4 @@
+.panel {
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.css]: is this being applied?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 8]
```diff
@@ -55,39 +61,37 @@ export class ConfigurationManagerComponent implements OnInit {
...
+
+          if (this.tree.treeModel.getFirstRoot()) {
+            this.tree.treeModel.getFirstRoot().setIsActive(true);
+            this.tree.treeModel.getFirstRoot().setIsExpanded(false);
+            this.tree.treeModel.getFirstRoot().setIsExpanded(true);
           }
         },
         error => {
-          /** request completed */
-          this.ngProgress.done();
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts]: we don't want ngprogress?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 9]
```diff
@@ -96,17 +100,48 @@ export class ConfigurationManagerComponent implements OnInit {
...
+    event.node.data.children = [];
+    if (event.node.isExpanded) {
+      this.configService.getChildren(event.node.data.id).
+        subscribe(
+          (data) => {
+            data['categories'].forEach(element => {
+              event.node.data.children.push({ id: element.key, name: element.description, hasChildren: true, children: [] });
+              this.tree.treeModel.update();
+            });
+            // event.node.data.children.push({ id: event.node.data.id, name: 'add', hasChildren: false, children: [] });
```
Comment (praveen-garg) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts]: remove this comment line?

File: src/app/components/core/configuration-manager/configuration-manager.component.ts [hunk 10]
```diff
@@ -171,6 +176,14 @@ export class ConfigurationManagerComponent implements OnInit {
     this.refreshCategory(categoryData.categoryKey, categoryData.categoryDescription);
   }
 
+  /**
+  * @param notify
```
Comment (monika-Dianomic) [inline src/app/components/core/configuration-manager/configuration-manager.component.ts]: No param in this method.

File: src/styles.css [hunk 11]
```diff
@@ -114,3 +114,16 @@ table.scroll tbody {
...
+    font-weight: bold;
+}
+
+.cat-desc a {
+    font-weight: bold;
+}
+
+.child-panel {
+    display: none;
+}
```
Comment (monika-Dianomic) [inline src/styles.css]: Is there need to add these css class?

**Pull request merged**